chain/stmgr/stmgr.go

```diff
@@ -146,12 +146,54 @@ type BlockMessages struct {
 
 type ExecCallback func(cid.Cid, *types.Message, *vm.ApplyRet) error
 
-func (sm *StateManager) ApplyBlocks(ctx context.Context, pstate cid.Cid, bms []BlockMessages, epoch abi.ChainEpoch, r vm.Rand, cb ExecCallback) (cid.Cid, cid.Cid, error) {
-	vmi, err := sm.newVM(pstate, epoch, r, sm.cs.Blockstore(), sm.cs.VMSys())
+func (sm *StateManager) ApplyBlocks(ctx context.Context, parentEpoch abi.ChainEpoch, pstate cid.Cid, bms []BlockMessages, epoch abi.ChainEpoch, r vm.Rand, cb ExecCallback) (cid.Cid, cid.Cid, error) {
+	vmi, err := sm.newVM(pstate, parentEpoch+1, r, sm.cs.Blockstore(), sm.cs.VMSys())
 	if err != nil {
 		return cid.Undef, cid.Undef, xerrors.Errorf("instantiating VM failed: %w", err)
 	}
 
+	runCron := func() error {
+		// TODO: this nonce-getting is a tiny bit ugly
+		ca, err := vmi.StateTree().GetActor(builtin.SystemActorAddr)
+		if err != nil {
+			return err
+		}
+
+		cronMsg := &types.Message{
+			To:       builtin.CronActorAddr,
+			From:     builtin.SystemActorAddr,
+			Nonce:    ca.Nonce,
+			Value:    types.NewInt(0),
+			GasPrice: types.NewInt(0),
+			GasLimit: build.BlockGasLimit * 10, // Make super sure this is never too little
```

Comment:
Aside: I think you want to set this way higher. We have little bandwidth to quantify this at present, and the consequences of hitting the limit are a lot worse than the consequences of a slow tipset.